examples/custom_metrics/model.py

```diff
@@ -0,0 +1,177 @@
+          * model_name: Model name
+        """
+
+        # Parse model_config and extract OUTPUT0 and OUTPUT1 configuration
+        self.model_config = model_config = json.loads(args['model_config'])
+        output0_config = pb_utils.get_output_config_by_name(
+            model_config, "OUTPUT0")
+        output1_config = pb_utils.get_output_config_by_name(
+            model_config, "OUTPUT1")
+
+        # Convert Triton types to numpy types
+        self.output0_dtype = pb_utils.triton_string_to_numpy(
+            output0_config['data_type'])
+        self.output1_dtype = pb_utils.triton_string_to_numpy(
+            output1_config['data_type'])
+
+        # Create a MetricFamily object to report the latency of the model
+        # execution. The 'kind' parameter must be either 'COUNTER' or
+        # 'GAUGE'.
+        # If duplicate name is used, both MetricFamily objects
+        # will reference to the same underlying MetricFamily. If there are two
+        # MetricFamily objects with the same name and same kind but different
+        # description, the original description will be used. Note that
+        # Duplicate name with different kind is not allowed.
+        self.metric_family = pb_utils.MetricFamily(
+            name="requests_process_latency_ns",
+            description="Cumulative time spent processing requests",
+            kind=pb_utils.COUNTER # or pb_utils.GAUGE
+        )
+
+        # Create a Metric object under the MetricFamily object. The 'labels'
+        # is a dictionary of key-value pairs. You can create multiple Metric
+        # objects under the same MetricFamily object with unique labels. Empty
+        # labels is allowed. The 'labels' parameter is optional. If you don't
+        # specify the 'labels' parameter, empty labels will be used.
+        self.metric = self.metric_family.Metric(
+            labels={"model" : "custom_metrics", "version" : "1"}
+        )
+
+    def execute(self, requests):
+        """`execute` MUST be implemented in every Python model. `execute`
+        function receives a list of pb_utils.InferenceRequest as the only
+        argument. This function is called when an inference request is made
+        for this model. Depending on the batching configuration (e.g. Dynamic
+        Batching) used, `requests` may contain multiple requests. Every
+        Python model, must create one pb_utils.InferenceResponse for every
+        pb_utils.InferenceRequest in `requests`. If there is an error, you can
+        set the error argument when creating a pb_utils.InferenceResponse
+
+        Parameters
+        ----------
+        requests : list
+          A list of pb_utils.InferenceRequest
+
+        Returns
+        -------
+        list
+          A list of pb_utils.InferenceResponse. The length of this list must
+          be the same as `requests`
+        """
+
+        output0_dtype = self.output0_dtype
+        output1_dtype = self.output1_dtype
+
```

Comment:
Not really related, but I think we should start getting rid of this copy/paste from PB examples and just reference `self.output0_dtype` and `self.output1_dtype` below directly. If it's a line length issue, we can just shorten the variable names a bit.

Reply:
Fixed.